### PR TITLE
Event loop precedence

### PR DESCRIPTION
**Pull request by Athi223**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Adds to #14

### CODE REVIEW

File: src/data/javascript-quiz.ts [hunk 1]
```diff
@@ -1676,6 +1676,17 @@ const javascriptQuiz = [
...
+  {
+    Question:
+      "In JavaScript, which one gets higher precendence among 'setTimeout' and 'Promise' (considering zero delay)?",
+    Answer: "Promise",
+    Distractor1: "setTimeout",
+    Distractor2: "Depends on the order of calling",
+    Distractor3: "Unpredictable",
+    Explanation:
+      "Promise always has higher precendence than setTimeout. The Promise will be executed before the setTimeout, irrespective of the order in which they are called.",
+    Link: "https://dmitripavlutin.com/javascript-promises-settimeout/",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1688]: please only use freeCodeCamp resources, official documentation or wikipedia articles for the link field

File: src/data/javascript-quiz.ts [hunk 2]
```diff
@@ -1676,6 +1676,17 @@ const javascriptQuiz = [
...
   },
+  {
+    Question:
+      "In JavaScript, which one gets higher precendence among 'setTimeout' and 'Promise' (considering zero delay)?",
+    Answer: "Promise",
+    Distractor1: "setTimeout",
+    Distractor2: "Depends on the order of calling",
+    Distractor3: "Unpredictable",
+    Explanation:
+      "Promise always has higher precendence than setTimeout. The Promise will be executed before the setTimeout, irrespective of the order in which they are called.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1687]:
```suggestion
      "Promises will always have a higher precedence than setTimeout. The Promise will be executed before the setTimeout, regardless of the order in which they are called.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

File: src/data/javascript-quiz.ts (hunk 1 above)
Comment (Athi223) [reply src/data/javascript-quiz.ts:1688]: Hey, sorry for linking an unofficial resource. To be honest it seemed to have some nice diagrams explaining the same, so linked it for the question. Will update the same.

Comment (Athi223) [reply src/data/javascript-quiz.ts]: Updated, please have a look

**Review by jdwilkin4 — APPROVED** [review]
Congrats on another contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**